### PR TITLE
[mono] Force Mono to respect explicit struct size when LayoutKind.Sequential is used

### PR DESCRIPTION
**Pull request by matouskozak**
We encountered an issue with Mono not respecting explicit size when `LayoutKind.Sequential` was used (https://github.com/dotnet/runtime/issues/101432). We fix this by eliminating addition of alignment padding when explicit size is set.

### CODE REVIEW

**Comment by grendello** [comment]
I didn't find any `StructLayout` use with explicit size in XA code (in-repo or generated), so we should be fine with that change.

`LibZipSharp` is fine too, however Mono.Posix has [one instance](https://github.com/mono/mono.posix/blob/main/src/Mono.Unix/Mono.Unix.Native/Syscall.cs#L1459-L1472) which might be affected?

**Comment by rolfbjarne** [comment]
> Is this change going to be a problem for your existing interop code in XI, XA or wasm?

XI doesn't have any sequential StructLayouts with explicit size, so it's good on our side.

Additionally we use CoreCLR for macOS, and we haven't found any such issues there.

**Comment by matouskozak** [comment]
> `LibZipSharp` is fine too, however Mono.Posix has [one instance](https://github.com/mono/mono.posix/blob/main/src/Mono.Unix/Mono.Unix.Native/Syscall.cs#L1459-L1472) which might be affected?

That one should be unaffected by this PR. Here we only modify the behavior for `LayoutKind.Sequential`. `LayoutKind.Explicit` was already handled correctly when used together with explicit size.

**Comment by lewing** [comment]
I'm not aware of any issues on the wasm side

**Comment by matouskozak** [comment]
We have tested this change on Arm64 VMs emulating Arm32 without any issues. We also performed successful local testing on armv7 Raspberry.

File: src/tests/Interop/ArrayMarshalling/UnalignedStructArray/MarshalUnalignedStructArray.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            new InnerStruct(1, 2),
+            new InnerStruct(3, 4),
+            new InnerStruct(5, 6),
+        };
+
+        fixed (InnerStruct* pStruct = &arrStructs[0])
+        {
+            byte* ptr = (byte*)pStruct;
+            ptr += 12;
+            Assert.Equal(3, *(int*)ptr);
```
Comment (lambdageek) [inline src/tests/Interop/ArrayMarshalling/UnalignedStructArray/MarshalUnalignedStructArray.cs:42]: I think this should be a `*(long*)ptr` - we want to check that 64-bit loads work.  the 32-bit load is aligned.

**Review by lambdageek — APPROVED** [review]
lgtm, thanks!

**Comment by matouskozak** [comment]
The WASM failures are unrelated and are matched using Build Analysis. The `tvos-arm64 Release AllSubsets_Mono` Tar failure is also known https://github.com/dotnet/runtime/issues/101692.

**Pull request merged**